src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -492,105 +511,138 @@ async function getGRPCInsecureCredential(frb: FunctionsRuntimeBundle): Promise<a
-
-    hasInitializedSettings = true;
-  };
+  const localAdminModule = require(adminResolution.resolution);
 
   adminModuleProxy = new Proxied<typeof admin>(localAdminModule)
-    .when("initializeApp", (adminModuleTarget) => (opts: any, appName: any) => {
+    .when("initializeApp", (adminModuleTarget) => (opts?: admin.AppOptions, appName?: string) => {
       if (appName) {
         new EmulatorLog("SYSTEM", "non-default-admin-app-used", "", { appName }).log();
         return adminModuleTarget.initializeApp(opts, appName);
       }
 
       const config = JSON.parse(process.env.FIREBASE_CONFIG || "{}");
-      new EmulatorLog("SYSTEM", "default-admin-app-used", `config=${config}`).log();
-
-      // TODO: Is there any possible harm in this?
       config.credential = makeFakeCredentials();
 
       if (frb.ports.database) {
         config.databaseURL = `http://localhost:${frb.ports.database}?ns=${frb.projectId}`;
         new EmulatorLog("SYSTEM", `Overriding database URL: ${config.databaseURL}`, "").log();
       }
 
-      app = adminModuleTarget.initializeApp({
+      const appOptions = {
         ...config,
         ...opts,
-      });
-      return app;
+      };
+      const originalApp = adminModuleTarget.initializeApp(appOptions);
+
+      new EmulatorLog("DEBUG", "default-admin-app-used", "", appOptions).log();
+      logDebug("Intializing default app.", appOptions);
+      defaultApp = proxyFirebaseApp(frb, originalApp, sslCreds);
+      return defaultApp;
     })
     .when("firestore", (adminModuleTarget) => {
-      const proxied = new Proxied<typeof admin.firestore>(adminModuleTarget.firestore);
-      return proxied
-        .applied(() => {
-          return new Proxied(adminModuleTarget.firestore())
-            .when("settings", () => {
-              return (settings: any) => {
-                initializeSettings(settings);
-              };
-            })
-            .any((target, field) => {
-              initializeSettings({});
-              return proxied.getOriginal(target, field);
-            })
-            .finalize();
-        })
-        .finalize();
+      // When we call admin.firestore() we want to forward this on to the default app,
+      // but if you access something like admin.firestore.FieldValue we don't want to
+      // do anything.
+      const adminFirestoreProxy = new Proxied<typeof admin.firestore>(
+        adminModuleTarget.firestore
+      ).applied(() => {
+        return defaultApp.firestore();
+      });
+
+      return adminFirestoreProxy.finalize();
     })
     .finalize();
 
-  require.cache[adminResolution] = {
+  // Stub the admin module in the require cache
+  require.cache[adminResolution.resolution] = {
     exports: adminModuleProxy,
   };
 
-  new EmulatorLog("DEBUG", "runtime-status", "firebase-admin has been stubbed.", {
+  logDebug("firebase-admin has been stubbed.", {
     adminResolution,
-  }).log();
+  });
+
   return adminModuleProxy;
 }
 
+function proxyFirebaseApp(
```

Comment:
Are there any changes to this logic?

Reply:
Yes.  The logic changed such that `app.firestore()` and `admin.firestore()` go to the same place.  PTAL.